SignalMessaging/environment/migrations/OWS104CreateRecipientIdentities.m

```diff
@@ -24,42 +24,36 @@ + (NSString *)migrationId
     return OWS104CreateRecipientIdentitiesMigrationId;
 }
 
-// Overriding runUp instead of runUpWithTransaction in order to implement a blocking migration.
-- (void)runUp
+- (void)runUpWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
 {
-    [[OWSRecipientIdentity dbReadWriteConnection] readWriteWithBlock:^(
-        YapDatabaseReadWriteTransaction *_Nonnull transaction) {
-        NSMutableDictionary<NSString *, NSData *> *identityKeys = [NSMutableDictionary new];
-
-        [transaction
-            enumerateKeysAndObjectsInCollection:TSStorageManagerTrustedKeysCollection
-                                     usingBlock:^(
-                                         NSString *_Nonnull recipientId, id _Nonnull object, BOOL *_Nonnull stop) {
-                                         if (![object isKindOfClass:[NSData class]]) {
-                                             OWSFail(
-                                                 @"%@ Unexpected object in trusted keys collection key: %@ object: %@",
-                                                 self.logTag,
-                                                 recipientId,
-                                                 object);
-                                             return;
-                                         }
-                                         NSData *identityKey = (NSData *)object;
-                                         [identityKeys setObject:identityKey forKey:recipientId];
-                                     }];
-
-        [identityKeys enumerateKeysAndObjectsUsingBlock:^(
-            NSString *_Nonnull recipientId, NSData *_Nonnull identityKey, BOOL *_Nonnull stop) {
-            DDLogInfo(@"%@ Migrating identity key for recipient: %@", self.logTag, recipientId);
-            [[[OWSRecipientIdentity alloc] initWithRecipientId:recipientId
-                                                   identityKey:identityKey
-                                               isFirstKnownKey:NO
-                                                     createdAt:[NSDate dateWithTimeIntervalSince1970:0]
-                                             verificationState:OWSVerificationStateDefault]
-                saveWithTransaction:transaction];
-        }];
-
-        [self saveWithTransaction:transaction];
+    NSMutableDictionary<NSString *, NSData *> *identityKeys = [NSMutableDictionary new];
+
+    [transaction
```

Comment:
Maybe I should add asserts in `OWSRecipientIdentity` that these entities are only fetched after the app is ready.

Reply:
> - I don't think the app should be considered "ready" until all the migrations have run.

Great point.  I've formalized this with two non-trivial changes (corresponding to two commits):

* be1fde905 Don't mark app as ready until all version migrations are done.
* a30170b3b Prefer "app is ready" flag to "storage is ready" flag.